### ISSUE TITLE
2.3.1 cannot upgrade database after sample data installation

### ISSUE THREAD

**Issue by rszczepaniec** [magento/magento2#23507]
Preconditions:
1. Clean Magento 2.3.1 community edition installation without sample data
2. PHP 7.2.12
3. mysql  Ver 15.1 Distrib 10.1.37-MariaDB

Steps to reproduce:
1. Install clean Magento 2.3.1 via composer
2. Enable developer mode
php bin/magento deploy:mode:set developer
3. Enable all modules
php bin/magento module:enable --all
4. Upgrade database, compile files, delopy static content, clear cache (until this moment work correctly)
php bin/magento setup:upgrade && php bin/magento setup:di:compile && php bin/magento setup:static-content:deploy en_US -f && php bin/magento indexer:reindex && php bin/magento cache:flush
5. Install Sample data
php bin/magento sampledata:deploy
6. Upgrade database
php bin/magento setup:upgrade

Expected result:
1. Correctly upgraded database

Actual result:
1. Cannot upgrade databse.
Console:
![file1](https://user-images.githubusercontent.com/29597016/60384762-3b21af00-9a82-11e9-82ff-652af32f3b8e.png)

system.log file:
![file2](https://user-images.githubusercontent.com/29597016/60384763-4248bd00-9a82-11e9-8777-419efd6aa3cc.png)

Excepltion thrown in Magento/Framework/Model/AbstractModel.php line 857
![file3](https://user-images.githubusercontent.com/29597016/60384770-5f7d8b80-9a82-11e9-9a51-d8c4eb585098.png)

**Comment by engcom-Bravo** [comment]
Hello @rszczepaniec. Thanks for reporting. We tried to reproduce Your issue exactly following the steps You provided, but we had no success. The process of upgrade had been lingering on Module 'Magento_CustomerSampleData' for a very long time compared to the other modules but after all the process went forward end everything successfully finished. No errors on that were found in system.log. By the way we did that as on Your reported version of 2.3.1 as well as on the version of 2.3.2. The results were identical. What exactly You mean under "Clean Magento"?  Is it a vanilla instance pulled from original source, having no code changed, added or removed? Or You mean it is simply clean of any products and other acquired entities?
